packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,787 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
+  static const String kConfigCodeSignCertificate = 'ios-signing-cert';
+
+  /// Config key for saved provisioning profile file path. A provisioning profile
+  /// sets criteria for who is allowed to sign code, what apps are allowed to be
+  /// signed, where and when those apps can be run and how those apps are entitled.
+  ///
+  /// Example: ~/Library/Developer/Xcode/UserData/Provisioning Profiles/1234567a-bcde-89f0-1234-g56hi567j8kl.mobileprovision
+  static const String kConfigCodeSignProvisioningProfile = 'ios-signing-profile';
+
+  /// Reset both [kConfigCodeSignCertificate] and [kConfigCodeSignProvisioningProfile]
+  /// config settings.
+  static void resetSettings(Config config, Logger logger) {
+    config.removeValue(kConfigCodeSignCertificate);
+    logger.printStatus('Removing "$kConfigCodeSignCertificate" value.');
+
+    config.removeValue(kConfigCodeSignProvisioningProfile);
+    logger.printStatus('Removing "$kConfigCodeSignProvisioningProfile" value.');
   }
 
-  // If the user's environment is missing the tools needed to find and read
-  // certificates, abandon. Tools should be pre-equipped on macOS.
-  final ProcessUtils processUtils = ProcessUtils(processManager: processManager, logger: logger);
-  if (!await processUtils.exitsHappy(const <String>['which', 'security']) ||
-      !await processUtils.exitsHappy(const <String>['which', 'openssl'])) {
-    return null;
+  /// Get and validate code-sign settings from the config (provisioning profile
+  /// or identity) or prompt the user to select a code-signing identity for
+  /// automatic code-signing.
+  ///
+  /// If a provisioning profile is saved, return settings with `CODE_SIGN_STYLE=Manual`,
+  /// `DEVELOPMENT_TEAM`, and `PROVISIONING_PROFILE_SPECIFIER`. If the profile
+  /// is no longer valid, return null.
+  ///
+  /// If a identity is saved, return settings with `DEVELOPMENT_TEAM`. If no
+  /// settings are found or if the saved identity is no longer valid, prompt
+  /// the user to select a code-signing identity for automatic code-signing.
+  ///
+  /// If [shouldExitOnNoCerts] is true, throw a toolExit if no code-signing
+  /// identities are found.
+  ///
+  /// If [automaticCodeSignStyleOnly] is true, return null if a provisioning
+  /// profile is saved in the config.
+  Future<Map<String, String>?> _getCodeSigningBuildSettings({
+    bool shouldExitOnNoCerts = true,
+    bool automaticCodeSignStyleOnly = false,
+  }) async {
+    if (!_platform.isMacOS) {
+      _logger.printTrace('Unable to get code-sign settings on non-Mac platform.');
+      return null;
+    }
+    final bool toolsValidated = await _validateCodeSignSearchTools();
+    if (!toolsValidated) {
+      return null;
+    }
+    final String? savedCertChoice =
+        _config.getValue(XcodeCodeSigningSettings.kConfigCodeSignCertificate) as String?;
+    final String? savedProfile =
+        _config.getValue(XcodeCodeSigningSettings.kConfigCodeSignProvisioningProfile) as String?;
```

Comment:
Nit: it looks like these aren't used until after code signing identities are found & validated. Consider moving this down. I don't feel strongly about this